src/DoctrineBridge/RecursiveIdFinder.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Iltar\HttpBundle\DoctrineBridge;
+
+use Iltar\HttpBundle\DoctrineBridge\Exception\CompositePrimaryKeyException;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessor;
+
+/**
+ * @author Iltar van der Berg <kjarli@gmail.com>
+ */
+final class RecursiveIdFinder implements IdFromEntityInterface
+{
+    /**
+     * @var RegistryInterface
+     */
+    private $registry;
+
+    /**
+     * @var PropertyAccessor
+     */
+    private $propertyAccessor;
+
+    /**
+     * @param RegistryInterface $registry
+     * @param PropertyAccessor  $propertyAccessor
+     */
+    public function __construct(RegistryInterface $registry, PropertyAccessor $propertyAccessor)
+    {
+        $this->registry = $registry;
+        $this->propertyAccessor = $propertyAccessor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getId($entity)
+    {
+        $className = get_class($entity);
+
+        $em = $this->registry->getManagerForClass($className);
+        $metadata = $em->getClassMetadata($className);
+
+        $identifier = $metadata->getIdentifier();
+
+        if (count($identifier) > 1) {
+            throw new CompositePrimaryKeyException(sprintf('Composite Primary Keys cannot be resolved to a single scalar value on Entity: %s', get_class($entity)));
+        }
+
+        $pkValue = $this->propertyAccessor->getValue($entity, $identifier[0]);
```

Comment:
IIRC, Doctrine uses reflection, it may be possible that the property accessor cannot access the ID property. I'm not sure if you want to change it though